File: backend/DevopsBankApi/DevopsBankApi/Models/Invoice.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DevopsBankApi.Models
{
    [Table("Invoice")]
    public partial class Invoice
    {
        [Column("id")]
        public long Id { get; set; }
        [Required]
        [StringLength(50)]
        public string InvoiceSender { get; set; }
        [Required]
        [StringLength(50)]
        public string RecipientName { get; set; }
        [Required]
        [Column("RecipientIBAN")]
        [StringLength(18)]
        public string RecipientIban { get; set; }
        [Required]
        [StringLength(20)]
        public string Reference { get; set; }
        [Required]
        [StringLength(10)]
        public string InvoiceNumber { get; set; }
        [Required]
        [Column("BIC")]
        [StringLength(8)]
        public string Bic { get; set; }
        [Column(TypeName = "decimal(18, 2)")]
        public decimal Total { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime DueDay { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime Date { get; set; }
    }
}

File: frontend/DevopsBank/src/app/models/invoice.ts
export class Invoice {
  id: number;
  invoiceSender: string;
  recipientName: string;
  recipientIban: string;
  reference: string;
  invoiceNumber: string;
  bic: string;
  total: number;
  dueDay: Date;
  date: Date;

  constructor(id?: number, invoiceSender?: string, recipientName?: string, recipientIban?: string, reference?: string,
              invoiceNumber?: string, bic?: string, total?: number, dueDay?: Date, date?: Date) {
    this.id = id;
    this.invoiceSender = invoiceSender;
    this.recipientName = recipientName;
    this.recipientIban = recipientIban;
    this.reference = reference;
    this.invoiceNumber = invoiceNumber;
    this.bic = bic;
    this.total = total;
    this.dueDay = dueDay;
    this.date = date;
  }
}
